util/parallel/parallel.go

```diff
@@ -0,0 +1,98 @@
+package parallel
+
+import (
+	"context"
+	"slices"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/codes"
+	"go.opentelemetry.io/otel/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+func New() parallelJobs {
+	return parallelJobs{}
+}
+
+func Run(ctx context.Context, name string, fn JobFunc) error {
+	return New().WithJob(name, fn).Run(ctx)
+}
+
+type parallelJobs struct {
+	Jobs  []Job
+	Limit *int
+}
+
+type Job struct {
+	Name string
+	Func JobFunc
+}
+
+type JobFunc func(context.Context) error
+
+func (p parallelJobs) WithJob(name string, fn JobFunc) parallelJobs {
+	p = p.Clone()
+	p.Jobs = append(p.Jobs, Job{name, fn})
+	return p
+}
+
+func (p parallelJobs) Clone() parallelJobs {
+	cp := p
+	cp.Jobs = slices.Clone(cp.Jobs)
+	return cp
+}
+
+func startSpan(ctx context.Context, name string) (context.Context, trace.Span) {
+	attr := []attribute.KeyValue{
+		attribute.Bool("dagger.io/ui.reveal", true),
+	}
+	return otel.
+		Tracer("dagger.io/util/parallel").
+		Start(ctx, name, trace.WithAttributes(attr...))
+}
+
+func (job Job) Runner(ctx context.Context) func() error {
+	ctx, span := startSpan(ctx, job.Name)
+	return func() error {
+		defer span.End()
+		if job.Func == nil {
```

Comment:
```suggestion
	return func() error {
		ctx, span := startSpan(ctx, job.Name)
		defer span.End()
		if job.Func == nil {
```

Without that change, the span can start but then not actually be running because the `errgroup` is waiting for a parallelism limit to unblock, which would be confusing and make the span look like it was taking much longer than it actually was.

Reply:
Ah! This is what I did initially, but it leads to the jobs being displayed in a random order every run, which I got comments about. Is there a way to get my cake and eat it too?

Reply:
Yeah not sure which is worse either 😭 

For now I'll prioritize getting this merged... Will add a FIXME.